src/components/Identity/createManualIdSyncs.js

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import createIdSyncs from "./createIdSyncs";
+import { isObject, isNumber, isNonEmptyString } from "../../utils";
+
+const createSyncIdByUrl = (
+  config,
+  logger,
+  cookieJar,
+  idSyncsProcessor
+) => data => {
+  let hasInvalidIdSyncs = false;
+
+  const idSyncs = (data.idSyncsArray || []).filter((idSync = {}) => {
+    if (
```

Comment:
I wonder if we could re-use the config validation code to perform this validation.  I'm working on the config stuff right now so it is top of mind, but I think this is something we could do in the future.